datafusion/core/src/config.rs

```diff
@@ -17,606 +17,490 @@
-    "datafusion.execution.parquet.pushdown_filters";
-
-/// Configuration option "datafusion.execution.parquet.reorder_filters"
-pub const OPT_PARQUET_REORDER_FILTERS: &str =
-    "datafusion.execution.parquet.reorder_filters";
-
-/// Configuration option "datafusion.execution.parquet.skip_metadata"
-pub const OPT_PARQUET_SKIP_METADATA: &str = "datafusion.execution.parquet.skip_metadata";
-
-/// Configuration option "datafusion.execution.parquet.metadata_size_hint"
-pub const OPT_PARQUET_METADATA_SIZE_HINT: &str =
-    "datafusion.execution.parquet.metadata_size_hint";
-
-/*-************************************
-*  Explain related
-**************************************/
-/// Configuration option "datafusion.explain.logical_plan_only"
-pub const OPT_EXPLAIN_LOGICAL_PLAN_ONLY: &str = "datafusion.explain.logical_plan_only";
-
-/// Configuration option "datafusion.explain.physical_plan_only"
-pub const OPT_EXPLAIN_PHYSICAL_PLAN_ONLY: &str = "datafusion.explain.physical_plan_only";
-
-/*-************************************
-*  Optimizer related
-**************************************/
-/// Configuration option "datafusion.optimizer.filter_null_join_keys"
-pub const OPT_FILTER_NULL_JOIN_KEYS: &str = "datafusion.optimizer.filter_null_join_keys";
-
-/// Configuration option "datafusion.optimizer.repartition_aggregations"
-pub const OPT_REPARTITION_AGGREGATIONS: &str =
-    "datafusion.optimizer.repartition_aggregations";
-
-/// Configuration option "datafusion.optimizer.repartition_joins"
-pub const OPT_REPARTITION_JOINS: &str = "datafusion.optimizer.repartition_joins";
-
-/// Configuration option "datafusion.optimizer.repartition_windows"
-pub const OPT_REPARTITION_WINDOWS: &str = "datafusion.optimizer.repartition_windows";
-
-/// Configuration option "datafusion.optimizer.skip_failed_rules"
-pub const OPT_OPTIMIZER_SKIP_FAILED_RULES: &str =
-    "datafusion.optimizer.skip_failed_rules";
-
-/// Configuration option "datafusion.optimizer.max_passes"
-pub const OPT_OPTIMIZER_MAX_PASSES: &str = "datafusion.optimizer.max_passes";
-
-/// Configuration option "datafusion.optimizer.top_down_join_key_reordering"
-pub const OPT_TOP_DOWN_JOIN_KEY_REORDERING: &str =
-    "datafusion.optimizer.top_down_join_key_reordering";
-
-/// Configuration option "datafusion.optimizer.prefer_hash_join"
-pub const OPT_PREFER_HASH_JOIN: &str = "datafusion.optimizer.prefer_hash_join";
-
-/// Configuration option "atafusion.optimizer.hash_join_single_partition_threshold"
-pub const OPT_HASH_JOIN_SINGLE_PARTITION_THRESHOLD: &str =
-    "datafusion.optimizer.hash_join_single_partition_threshold";
-
-/// Configuration option "datafusion.execution.round_robin_repartition"
-pub const OPT_ENABLE_ROUND_ROBIN_REPARTITION: &str =
-    "datafusion.optimizer.enable_round_robin_repartition";
+use datafusion_common::{DataFusionError, Result};
+use std::any::Any;
+use std::collections::BTreeMap;
+use std::fmt::Display;
+
+macro_rules! config_namespace {
```

Comment:
👍 

I think some introduction doc comments here would help -- specifically some examples of what this macro is creating would be nice